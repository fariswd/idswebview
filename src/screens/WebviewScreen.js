import React from 'react'
import {
  View,
  Text,
  ActivityIndicator,
} from 'react-native'
import { WebView } from 'react-native-webview';
import Loading from '../components/Loading'

import Colors from '../constants/Colors'

export default class WebviewScreen extends React.Component {
  render(){
    return (
      <View style={{flex: 1}}>
        <WebView
          source={{ uri: 'https://www.indonesiadefenceservices.com/' }}
          startInLoadingState={true}
          renderLoading={() => <Loading />}
        />
      </View>
    )
  }
}