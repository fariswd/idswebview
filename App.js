import React from 'react';
import {
  SafeAreaView,
  View,
  StatusBar,
} from 'react-native';

import WebviewScreen from './src/screens/WebviewScreen'
import SplashScreen from './src/screens/SplashScreen';
import TosScreen from './src/screens/TosScreen';
import helpers from './src/utils/helpers'

export default class App extends React.Component {
  constructor(){
    super()
    this.state = {
      splash: false,
      showWebview: false,
    }
  }
  componentDidMount(){
    this.hideSplashScreen()
  }
  hideSplashScreen(){
    setTimeout(() => {
      this.setState({splash: true});
    }, 2000);
  }
  toWebview(){
    this.setState({ showWebview: true })
  }
  render(){
    const { splash, showWebview } = this.state
    if(!splash){
      return <SplashScreen />
    } else {
      return (
        <View style={{flex: 1}}>
          {!helpers.isAndroid && <StatusBar barStyle="dark-content"/>}
          <SafeAreaView style={{ flex: 1 }}>
            {showWebview
              ? <WebviewScreen />
              : <TosScreen buttonFn={() => this.toWebview()} />
            }
          </SafeAreaView>
        </View>
      );
    }
  }
}
