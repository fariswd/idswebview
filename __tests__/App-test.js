/**
 * @format
 */

import 'react-native'
import { TouchableOpacity } from 'react-native'
import React from 'react'
import App from '../App'
import TosScreen from '../src/screens/TosScreen'
import WebviewScreen from '../src/screens/WebviewScreen'
import Enzyme, {shallow} from 'enzyme'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

jest.useFakeTimers()

test('render app', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('hide splashscreen', () => {
  const methodHide = jest.spyOn(App.prototype, 'hideSplashScreen');
  const wrapper = shallow(<App />)
  expect(methodHide).toHaveBeenCalled();
})

test('press on touchable opacity', () => {
  const onPressEvent = jest.fn()
  onPressEvent.mockReturnValue('Link on press invoked')
  const wrapper = shallow(
    <TosScreen buttonFn={onPressEvent} />,
  )
  wrapper.find(TouchableOpacity).first().props().onPress()
  expect(onPressEvent.mock.calls.length).toBe(1)
})

test('render webview', () => {
  const tree = renderer.create(<WebviewScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})
