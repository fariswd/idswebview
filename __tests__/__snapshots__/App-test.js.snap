// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://assets.website-files.com/5a1a80a445a0eb000187a6ee/5a5e17799629620001d0331d_IDS%20LOGO%20web.gif",
      }
    }
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</View>
`;

exports[`render webview 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onNavigationStateChange={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "uri": "https://www.indonesiadefenceservices.com/",
        }
      }
      startInLoadingState={true}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#eb4034"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
